### PR TITLE
Scc 3772/page component tests

### PR DESCRIPTION
**Pull request by charmingduchess**
Create tests for existing pages and components. Some tests could not be implemented, and is pending 500 error messaging implementation, and making the search form a controlled component (tickets forthcoming).

### CODE REVIEW

File: __test__/pages/advancedSearchForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    screen.getByText(
+      "Please enter at least one field to submit an advanced search."
+    )
+  })
+  it("can set keyword, contributor, title, subject", async () => {
+    render(<AdvancedSearch />)
+
+    const [keywordInput, contributorInput, titleInput, subjectInput] =
+      screen.getAllByRole("textbox")
+    await act(async () => {
```
Comment (EdwinGuzman) [inline __test__/pages/advancedSearchForm.test.tsx:36]: The `await` on the per userEvent line is okay, but you shouldn't need this `act(...)` function wrapper.

File: __test__/pages/advancedSearchForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    })
+  })
+  it("can select languages", async () => {
+    render(<AdvancedSearch />)
+
+    const languageSelect = screen.getByRole("combobox", { name: "Language" })
+    await act(async () => {
+      await userEvent.selectOptions(languageSelect, "Azerbaijani")
+      submit()
+      expect(mockRouter.asPath).toBe(
```
Comment (EdwinGuzman) [inline __test__/pages/advancedSearchForm.test.tsx:54]: Add a comment that the language code for Azerbaijani is "aze" and that is expected in the url query param.

File: __test__/pages/advancedSearchForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    })
+  })
+  it("can check material checkboxes", async () => {
+    render(<AdvancedSearch />)
+    await act(async () => {
+      await userEvent.click(screen.getByLabelText("Notated music"))
+      await userEvent.click(screen.getByLabelText("Cartographic"))
+      submit()
+      expect(mockRouter.asPath).toBe(
+        "/search?q=&filters%5BmaterialType%5D%5B0%5D=resourcetypes%3Anot&filters%5BmaterialType%5D%5B1%5D=resourcetypes%3Acar"
```
Comment (EdwinGuzman) [inline __test__/pages/advancedSearchForm.test.tsx:66]: To make sure, "not" and "car" are the codes for "Notated Music" and "Cartographic", right?

**Review by EdwinGuzman — COMMENTED** [review]
Overall looks good but just wondering about the outer `await act(...` calls you have. The await for the userEvents should already wait until they're done before moving on so try removing them.

File: __test__/pages/advancedSearchForm.test.tsx (hunk 1 above)
Comment (charmingduchess) [reply __test__/pages/advancedSearchForm.test.tsx:36]: fixed

File: __test__/pages/advancedSearchForm.test.tsx (hunk 2 above)
Comment (charmingduchess) [reply __test__/pages/advancedSearchForm.test.tsx:54]: added

File: __test__/pages/advancedSearchForm.test.tsx (hunk 3 above)
Comment (charmingduchess) [reply __test__/pages/advancedSearchForm.test.tsx:66]: added

**Comment by charmingduchess** [comment]
> Overall looks good but just wondering about the outer `await act(...` calls you have. The await for the userEvents should already wait until they're done before moving on so try removing them.

When I removed the await, the tests appeared to pass, but I could not make them fail. I think the await is necessary.

File: __test__/pages/404.test.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+describe("404", () => {
+  it("should display 404 text", () => {
+    render(<Custom404 />)
+
+    const FourOhFourText = "404 Not Found"
+    const heading = screen.getByRole("heading")
+    expect(heading).toHaveTextContent(FourOhFourText)
+  })
+  it("should have links to circ and legacy catalogs", () => {
```
[the remark this answers is not in the document]
Comment (charmingduchess) [reply __test__/pages/404.test.tsx]: finished!

**Comment by charmingduchess** [comment]
I got the next config to recognize the pages file extension, so I was able to move all tests next to the components they test.

**Comment by EdwinGuzman** [comment]
> When I removed the await, the tests appeared to pass, but I could not make them fail. I think the await is necessary.

Gotcha, I'm surprise you need this much async structure for running the tests but it makes sense that some commands, like typing, need to wait to be done.

**Comment by EdwinGuzman** [comment]
The update looks good. I dont want to approve yet because the other PR should be merged first, or point this PR to that PR. There will be some conflicts either way that should be resolved first.

**Pull request merged**